src/main/java/org/lobid/resources/DirReader.java

```diff
@@ -0,0 +1,66 @@
+package org.lobid.resources;
+
+import java.io.File;
+
+import org.culturegraph.mf.exceptions.MetafactureException;
+import org.culturegraph.mf.framework.DefaultObjectPipe;
+import org.culturegraph.mf.framework.ObjectReceiver;
+import org.culturegraph.mf.framework.annotations.Description;
+import org.culturegraph.mf.framework.annotations.In;
+import org.culturegraph.mf.framework.annotations.Out;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Reads a directory and emits all filenames found.
+ * 
+ * @author Markus Michael Geipel, Pascal Christoph
+ */
+@In(String.class)
+@Out(String.class)
+@Description("Reads a directory and emits all filenames found.")
+public final class DirReader
```

Comment:
We have `DirReader` in metafacture-core:
https://github.com/hbz/metafacture-core/blob/master/src/main/java/org/culturegraph/mf/stream/source/DirReader.java